app/api_topology.go

```diff
@@ -114,7 +114,11 @@ func handleWebsocket(
 	var (
 		previousTopo render.RenderableNodes
 		tick         = time.Tick(loop)
+		wait         = make(chan struct{}, 1)
 	)
+	rep.WaitOn(wait)
+	defer rep.UnWait(wait)
```

Comment:
I'm debating if this would be any cleaner with a watch and lock. instead of a channel. Not sure...